examples/datepicker/js/datepicker.js

```diff
@@ -0,0 +1,809 @@
+  }
+};
+
+DatePicker.prototype.handlePreviousYearButton = function (event) {
+  var flag = false;
+  var onFirstRow = this.onFirstRow();
+  var onLastRow = this.onLastRow();
+
+  switch (event.type) {
+
+    case 'keydown':
+
+      switch (event.keyCode) {
+
+        case this.keyCode.RETURN:
+        case this.keyCode.SPACE:
+          this.moveToPreviousYear();
+          this.adjustCurrentDay(onFirstRow, onLastRow);
+          this.setFocusDay(false);
+          flag = true;
+          break;
+
+        case this.keyCode.TAB:
+          if (event.shiftKey) {
+            this.okButtonNode.focus();
+            flag = true;
+          }
+          break;
+
+        case this.keyCode.ESC:
+          this.hide();
+          flag = true;
+          break;
+
+        default:
+          break;
+      }
+
+      break;
+
+    case 'click':
+      this.moveToPreviousYear();
+      this.adjustCurrentDay(onFirstRow, onLastRow);
+      this.setFocusDay(false);
+      break;
+
+    default:
+      break;
+  }
+
+  if (flag) {
+    event.stopPropagation();
+    event.preventDefault();
+  }
+};
+
+DatePicker.prototype.handleNextMonthButton = function (event) {
+  var flag = false;
+  var onFirstRow = this.onFirstRow();
+  var onLastRow = this.onLastRow();
+
+  switch (event.type) {
+
+    case 'keydown':
+
+      switch (event.keyCode) {
+        case this.keyCode.ESC:
+          this.hide();
+          flag = true;
+          break;
+
+        case this.keyCode.RETURN:
+        case this.keyCode.SPACE:
+          this.moveToNextMonth();
+          this.adjustCurrentDay(onFirstRow, onLastRow);
+          this.setFocusDay(false);
+          flag = true;
+          break;
+      }
+
+      break;
+
+    case 'click':
+      this.moveToNextMonth();
+      this.adjustCurrentDay(onFirstRow, onLastRow);
+      this.setFocusDay(false);
+      break;
+
+    default:
+      break;
+  }
+
+  if (flag) {
+    event.stopPropagation();
+    event.preventDefault();
+  }
+};
+
+DatePicker.prototype.handlePreviousMonthButton = function (event) {
+  var flag = false;
+  var onFirstRow = this.onFirstRow();
+  var onLastRow = this.onLastRow();
+
+  switch (event.type) {
+
+    case 'keydown':
+
+      switch (event.keyCode) {
+        case this.keyCode.ESC:
+          this.hide();
+          flag = true;
+          break;
+
+        case this.keyCode.RETURN:
+        case this.keyCode.SPACE:
+          this.moveToPreviousMonth();
+          this.adjustCurrentDay(onFirstRow, onLastRow);
+          this.setFocusDay(false);
+          flag = true;
+          break;
+      }
+
+      break;
+
+    case 'click':
+      this.moveToPreviousMonth();
+      this.adjustCurrentDay(onFirstRow, onLastRow);
+      this.setFocusDay(false);
+      flag = true;
+      break;
+
+    default:
+      break;
+  }
+
+  if (flag) {
+    event.stopPropagation();
+    event.preventDefault();
+  }
+};
+
+DatePicker.prototype.moveToNextYear = function () {
+  this.year++;
+  this.updateGrid();
```

Comment:
Setting a single date cell to `tabindex="0"` should probably be handled in the `updateGrid` function instead of `updateFocus`, since currently moving to the next or previous year or month only calls `updateGrid`, which then makes the date grid inaccessible via the keyboard.

Reply:
The same cell has focus after moving by month or year, so no need to update the focus or the `tabindex` value